crates/gitbutler-branch-actions/tests/fixtures/dependencies.sh

```diff
@@ -343,4 +397,45 @@ update 8
 added at the bottom
 " > file
   $CLI branch commit my_stack -m "insert 1 line at the top and bottom, remove lines 3 and 4 and update line 7"
-)
\ No newline at end of file
+)
+
+git clone remote complex-branch-checkout
+(cd complex-branch-checkout
+  git config user.name "Author"
+  git config user.email "author@example.com"
+
+  git switch -c my_stack
+  echo "this is a" > a
+  git add a && git commit -m "add a" --trailer ""
+  set_change_id "$(git rev-parse HEAD)" "change-id-1" "my_stack"
```

Comment:
We need to set the change ID in the local branch commits in order to maintain the 'complex' structure.

If the commits don't have a change ID, they'll be rebased so that they all get one. See https://github.com/gitbutlerapp/gitbutler/pull/5636

This process will get rid of all but the first parent of every commit and skip empty commits, effectively loosing the `"delete b"` commit, which we want to keep